src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AContainer.java

```diff
@@ -243,15 +244,16 @@ protected void tick(Block b) {
             if (timeleft > 0) {
                 ChestMenuUtils.updateProgressbar(inv, 22, timeleft, processing.get(b).getTicks(), getProgressBar());
 
-                if (isChargeable()) {
-                    if (getCharge(b.getLocation()) < getEnergyConsumption()) {
-                        return;
-                    }
-
-                    removeCharge(b.getLocation(), getEnergyConsumption());
+                if (!takeCharge(b.getLocation())) {
+                    return;
                 }
+
                 progress.put(b, timeleft - 1);
             } else {
+                if (!takeCharge(b.getLocation())) {
+                    return;
+                }
```

Comment:
No need for this much nesting and unnecessary guard blocks.

```suggestion
            } else if (takeCharge(b.getLocation())) {
```